google/cloud/opentelemetry/internal/time_series.cc

```diff
@@ -0,0 +1,49 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/opentelemetry/internal/time_series.h"
+#include "google/cloud/opentelemetry/internal/monitored_resource.h"
+#include "google/cloud/internal/absl_str_replace_quiet.h"
+#include "absl/types/variant.h"
+#include <opentelemetry/common/attribute_value.h>
+
+namespace google {
+namespace cloud {
+namespace otel_internal {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+
+google::api::Metric ToMetric(
+    opentelemetry::sdk::metrics::MetricData const& metric_data,
+    opentelemetry::sdk::metrics::PointAttributes const& attributes) {
+  google::api::Metric proto;
+  auto const metric_type =
+      "workload.googleapis.com/" + metric_data.instrument_descriptor.name_;
```

Comment:
This hardcoded prefix is suspect.  And I believe maybe it does not work for custom metrics:

https://cloud.google.com/monitoring/api/v3/naming-conventions

That suggests all metrics created by the application go to `custom.googleapis.com`.

Reply:
The other exporters use `workload.googleapis.com`. That is what is in: go/otel-gcp-metric-exporter-spec

> hardcoded

It is configurable in other exporters. I am not there yet.

> maybe it does not work for custom metrics

It does work. I have tried it. https://screenshot.googleplex.com/94MR7j6wFz7frnA